TESTING.md

```diff
@@ -0,0 +1,119 @@
+# Testing
+
+Tools used to test the application:
+
+* **ESLint** (static code analysis of JS files)
+* **Markdownlint** (static analysis of MD files)
+* **Stylelint** (static code analysis of CSS files)
+* **Jest** (unit testing)
+* **Playwright** (visual and functional component testing)
+
+Generally, `npm test` and `npm run test:playwright-ct:all` should be run  within
+their designated Docker containers before pushing changes to the repository.
+
+## Tools
+
+### Linters (ESlint, Markdownlint, Stylelint)
+
+**On host:**
+
+[Open][gh-gg-node-shell] `node_shell` Docker container:
+
+```bash
+docker compose run --rm node_shell
+```
+
+**Within `node_shell`:**
+
+Run linters either all together:
+
+```bash
+npm run lint
+```
+
+or run linters individually:
+
+```bash
+npm run <eslint|markdownlint|stylelint>
+```
+
+### Jest
+
+**On host:**
+
+[Open][gh-gg-node-shell] `node_shell` Docker container:
+
+```bash
+docker compose run --rm node_shell
+```
+
+**Within `node_shell`:**
+
+Run Jest tests:
+
+```bash
+npm run test:jest
+```
+
+### Playwright
+
+Playwright tests must be run in a Docker container to ensure a uniform
+environment. Otherwise, snapshots would differ between operating systems.
+
+This is the reason why you need to run Playwright tests separately
+from other tools.
+
+#### Configuration
+
+Test parameters can be tweaked by creating and tweaking `.env.playwright` file:
+
+```bash
+cp .env.playwright.dist .env.playwright
+```
+
+#### Running Tests
+
+Playwright tests can be run using the following commands:
+
+**On host:**
+
+[Open][gh-gg-playwright] `playwright` Docker container:
+
+```bash
+docker compose run --rm playwright
+```
+
+**Within `node_shell`:**
+
+Run tests:
+
+```bash
+npm run test:playwright-ct:<all|all-with-update>
+```
+
+You can also run specific tests by passing a path to the test files:
+
+```bash
+npm run test:playwright-ct:<all|all-with-update> -- <match_path>
+```
+
+You can also pass any [CLI command][playwright-cli] to the test runner:
+
+```bash
+npm run test:playwright-ct:<all|all-with-update> -- <cli_argument>
+```
+
+#### Opening Test Report
+
+After running Playwright tests, test report can be opened **on the host system**
+by using the following command:
+
+```bash
+npm run test:playwright-ct:show-report
+```
+
+Test report cannot be server from Docker container yet.
+
+[gh-gg-node-shell]: /docs/contribute/general-guidelines#node-shell
```

Comment:
These links do not work outside of the context of mkdocs. We need to add `src` to the path.

I dislike having this text twice in the repo, but this also goes for `CONTRIBUTING.md` so I think it it beyond the scope of this issue -> I will raise it elsewhere.